chacra/controllers/repos/projects.py

```diff
@@ -0,0 +1,55 @@
+from pecan import expose, abort, request
+from chacra.models import Project
+from chacra.models.repos import Repo
+from chacra import models
+from chacra.controllers import error
+from chacra.controllers.repos.refs import RefController
+
+
+class ProjectController(object):
+
+    def __init__(self, project_name):
+        self.project_name = project_name
+        self.project = Project.query.filter_by(
+            name=project_name
+        ).join(Repo).filter(
+            Repo.path != None
+        ).first()
+        if not self.project:
+            if request.method != 'POST':
+                abort(404)
+            elif request.method == 'POST':
```

Comment:
I am not sure we should allow this specific situation. Unless we want to allow `/repos/` to have disconnected relationships with `/projects/`. For example if I create repos without pushing binaries and bypassing the whole automatic provisioning of repositories